src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -89,7 +89,7 @@ public FunctionDescriptor(FunctionDescriptorParams funcDescParams)
                 {
                     x.Function = this;
                     return x;
-                });
+                }).ToList();
```

Comment:
Why converting it to `List` here? Because we have `Any()` and `Select()` operation on it. If this `IEnumerable` contains some lazy evaluation, we have to do the real evaluation twice for the first element.